numpy/lib/financial.py

```diff
@@ -597,6 +612,9 @@ def irr(values):
     that gives a net present value of 0.0; for a more complete explanation,
     see Notes below.
 
+    Decimal type is not supported for values due to the use of square roots within
+    the calculation(s).
```

Comment:
Really? `np.sqrt(Decimal(2))` works for me.

Reply:
I misrepresented this statement. It's because of the call to np.roots and it returning imaginary numbers. 

Currently I do not think the underlying `eigvals` call within the np.roots method will work because the polynomial call in `np.roots` does a force casts to floats. 

I've given it a try to not force cast to float, and keep the `object` type for dtype. This causes later problems inside of the `eigvals` call (called form `np.roots`). These lines do not support `Decimal` and I'm not sure how to add it: https://github.com/numpy/numpy/blob/v1.13.3/numpy/linalg/linalg.py#L894-L895, there is no matching signature that can be called.

Reply:
Yeah, sounds like you're on a loser there. Maybe just fix the comment to refer to `` `roots` ``.

Reply:
I've not logged a ticket for this issue because I think it requires a lot more work than a single ticket to end up being supported, and relates heavily to the effort to get `Decimal` to have greater support within numpy overall.